VERSIONING.md

```diff
@@ -0,0 +1,55 @@
+Rocket Chip Versioning Strategy ** All Strawman to be discussed**
+===============================
+
+Version Naming
+--------------
+
+* Rocket Chip uses a Calendar Versioning release strategy (https://calver.org/).
+* The format is YYYY.MM.DD.MINOR
+* MINOR changes include:
+  * API Additions
+  * Bug Fixes (which can result in changes in the emitted hardware considered bugs)
+* MINOR changes do NOT include:
+  * Changes in top-level I/Os for < level of hierarchy >
+  * Backwards incompatible changes in the API (see below)
+
+API
+---
+
+The Rocket-Chip API is: 
+  * All public Scala methods and classes
+  * All public Scala methods and classes of included Scala libraries (hardfloat, CDE, etc)
```

Comment:
It isn't clear to me if the definition of the Scala API implies either source or binary compatibility for minor version bumps (the discussion below about `public val`s in `Module`s seems to imply neither, which is a pragmatic choice but differs from how most scala projects work). I think it would be good to be specific as to whether one of (or any of) binary and source compatibility are intended to be preserved.

I see that mima is in plugins.sbt- is the intention for CI to report binary compatibility for PRs? That would be awesome, even if it can't always be preserved.

Reply:
> It isn't clear to me if the definition of the Scala API implies either source or binary compatibility for minor version bumps (the discussion below about public vals in Modules seems to imply neither, which is a pragmatic choice but differs from how most scala projects work).

Yeah, this is a big question in my mind. On the one hand, just using "normal" Scala API makes a lot of sense, since in actuality every `val foo = Wire(...)` you create really is an API point for someone else even in scala-land (there are utilities to leverage that to generate cross-module references, etc).

On the other hand, the thing that often causes/cures pain for Chisel consumers is things like:
`val foo = Reg(...).suggestName("a_better_name")`
because it gives an API to the *output* of the generated code.
"a_better_name"  becomes a pretty important part of the API too, but will be harder to detect with binary compatibility.

I'll be honest that I am just getting familiarized with this topic myself, and would be interested in learning more how this binary compatability stuff should be done.